internal/config/auth.go

```diff
@@ -36,7 +36,7 @@ func schemeHostOnly(longURL string) (string, error) {
 }
 
 func GetBaseURL() string {
-	if endpoint := os.Getenv("DATAROBOT_ENDPOINT"); endpoint != "" {
+	if endpoint := viper.GetString("endpoint"); endpoint != "" {
```

Comment:
Is `viper.GetString("endpoint")` guaranteed to get value from `DATAROBOT_ENDPOINT` env var first?
If yes, then use `DataRobotURL` constant here and remove two identical if statements.
Otherwise, revert with comment:

```suggestion
	// Get endpoint from environment variable when running in codespace
	if endpoint := os.Getenv("DATAROBOT_ENDPOINT"); endpoint != "" {
```

Reply:
BLUF, yes, env takes precedence.

---

> Viper uses the following precedence order. Each item takes precedence over the item below it:
> 
> explicit call to Set
> flag
> env
> config
> key/value store
> default

https://pkg.go.dev/github.com/spf13/viper#section-readme

Because we do not call `SetString("endpoint", x)` and `endpoint` is not a CLI flag, yes, env will always take precedence.

---
> One important thing to recognize when working with ENV variables is that the value will be read each time it is accessed. Viper does not fix the value when the BindEnv is called.

https://pkg.go.dev/github.com/spf13/viper#readme-working-with-environment-variables

---
```
╰─ ❯ cat ~/.config/datarobot/drconfig.yaml | grep endpoint
endpoint: https://staging.datarobot.com/api/v2

╰─ ❯ dist/dr version | grep endpoint
  endpoint: https://staging.datarobot.com/api/v2

╰─ ❯ DATAROBOT_ENDPOINT=hey dist/dr version | grep endpoint
  endpoint: hey

╰─ ❯ DATAROBOT_API_ENDPOINT=no DATAROBOT_ENDPOINT=hey dist/dr version | grep endpoint
  endpoint: hey

╰─ ❯ DATAROBOT_ENDPOINT=hey DATAROBOT_API_ENDPOINT=no dist/dr version | grep endpoint
  endpoint: hey

╰─ ❯ export DATAROBOT_ENDPOINT=hey                   
╰─ ❯ DATAROBOT_API_ENDPOINT=no dist/dr version | grep endpoint
  endpoint: hey

╰─ ❯ unset DATAROBOT_ENDPOINT      
╰─ ❯  dist/dr version | grep endpoint
  endpoint: https://staging.datarobot.com/api/v2

╰─ ❯ DATAROBOT_API_ENDPOINT=r-says-hey dist/dr version | grep endpoint
  endpoint: r-says-hey
```